homeassistant/components/kodi/media_player.py

```diff
@@ -435,6 +439,26 @@ async def ws_loop_wrapper():
         # run until the websocket connection is closed.
         self.hass.loop.create_task(ws_loop_wrapper())
 
+    async def async_added_to_hass(self):
+        """Connect the websocket if needed."""
+        if not self._enable_websocket:
+            return
+
+        await self._async_connect_websocket_if_disconnected()
```

Comment:
Why is there a difference between connecting in `async_added_to_hass` compared to in `async_update`? Both will run during entity addition.

Reply:
Based on the comments in the issues (https://github.com/home-assistant/core/issues/38088#issuecomment-663476923), I *think* what is happening is the websocket can't connect so it times out and `async_update` gives us a chain of connection attempts via `await self._ws_server.ws_connect()` that are always trying to connect and thus blocking startup.



Since I don't have a `kodi` device, I'm not 100% sure that is what is going on.

Reply:
Not sure if it's related but it's weird that we pass a coroutine object of the state update when creating the turn on script instance. I'd expect that we pass a method as target, not the coroutine instance.

Reply:
Its possible that could create another connect attempt but shouldn't be an issue anymore since we don't try to reconnect every 15 seconds now from `async_update`

Reply:
I changed it to `self.hass.loop.create_task(self.async_ws_connect())`